Packs/CiscoSpark/Integrations/CiscoWebexEventCollector/CiscoWebexEventCollector.py

```diff
@@ -0,0 +1,536 @@
+        }
+
+    def oauth_test(self):
+        """Runs the `get_compliance_officer_events` method in order to tes the connection."""
+        self.get_compliance_officer_events(date_time_to_iso_format(datetime.utcnow() - timedelta(hours=3)))
+
+    def get_compliance_officer_events(self, from_date: str, limit: int = DEFAULT_MAX_FETCH,
+                                      next_url: str = '') -> requests.Response:
+        """
+        Returns events either with a `next_url` or according to filter parameters.
+        Args:
+            from_date: A string represents a datetime is ISO format from when to get the events.
+            limit: A number of how mny events to return.
+            next_url: A URL (returned from the previous run) to get the events (using for pagination).
+
+        Returns:
+            A response object with the events returned dform the API.
+        """
+        if next_url:
+            return self._http_request(method='GET', full_url=next_url, resp_type='response')
+        params = {
+            'from': from_date,
+            'to': date_time_to_iso_format(datetime.utcnow()),
+            'max': min(limit, 1000),
+        }
+        return self._http_request(method='GET', url_suffix='events', params=params, resp_type='response')
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module():
+    """Raises an exception with the message to run the `!cisco-webex-oauth-start` command."""
+    raise DemistoException(
+        'In order to authorize the instance, first run the command `!cisco-webex-oauth-start`, '
+        'and complete the process in the URL that is returned. You will then be redirected '
+        'to the callback URL. Copy the authorization code found in the query parameter '
+        '`code`, and paste that value in the command `!cisco-webex-oauth-complete` as an argument to finish '
+        'the process. Then you can test it bu running the `!cisco-webex-oauth-test` command.'
+    )
+
+
+def oauth_start(client: Client) -> CommandResults:
+    """
+    Runs the client `oauth_start` method in order to create a URL to start the authenticate process.
+    Args:
+        client: A client object either an AdminClient or a ComplianceOfficerClient.
+
+    Returns:
+        A CommandResult with a URL generated according to the client attributes to start the authentication.
+    """
+    url, user = client.oauth_start()
+    message = f"""
+>### Authorization instructions
+>1. Click on the [login URL]({url}) to sign in and grant Cortex XSOAR permissions for your Cisco Webex {user} application.
+You will be automatically redirected to a link with the following structure:
+```REDIRECT_URI?code=AUTH_CODE```
+>2. Copy the `AUTH_CODE` (without the `code=` prefix)
+and use it in **!cisco-webex-oauth-complete** command as a value fot the **code** argument.
+"""
+    return CommandResults(readable_output=message)
+
+
+def oauth_complete(client: Client, args: dict) -> CommandResults:
```

Comment:
since you have 2 clients in this class its better in the hint of client to specify exactly which client type you expect here Admin / ComplianceOfficer 
**change in all methods calling client: Client**

Reply:
This function is actually a generic function for both clients so I think it's better to set to `Client` instead of `AdminClient | ComplianceOfficerClient`

Reply:
Yes if it is called for both clients you are right